sql/src/main/java/io/crate/execution/dsl/projection/WindowAggProjection.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.execution.dsl.projection;
+
+import com.google.common.collect.ImmutableMap;
+import io.crate.analyze.WindowDefinition;
+import io.crate.expression.symbol.Symbol;
+import io.crate.expression.symbol.Symbols;
+import io.crate.expression.symbol.WindowFunction;
+import io.crate.planner.ExplainLeaf;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class WindowAggProjection extends Projection {
+
+    private final WindowDefinition windowDefinition;
+    private final List<Symbol> standaloneWithInputs;
+    private LinkedHashMap<WindowFunction, List<Symbol>> functionsWithInputs;
+    private ArrayList<Symbol> outputs;
+
+    public WindowAggProjection(WindowDefinition windowDefinition,
+                               LinkedHashMap<WindowFunction, List<Symbol>> functionsWithInputs,
```

Comment:
Wanted to make sure we receive a LinkedHashMap so we maintain the order of the functions when we execute them. 

eg. in a case of `select sum(x) over(), avg(x) over() from t`, if a regular hashmap is used there's no deterministic way of ensuring the result is in the form of `sum | avg`

Reply:
It is exposed as a HashMap as the consumers don't need to know this, they can iterate the entries as they would with any *Map

Reply:
I'd prefer to just require the interface this component need. The requirement of a LinkedHashMap is outside of this component.

Reply:
But this makes it explicit to the `WindowAggProjection` creators. The alternative is for this to be a Map in the constructor and add an assertion inside the constructor that the map is a LinkedHashMap (otherwise the creators could pass in a HashMap), but I'd prefer the explicit way